File: leetcode/349-intersection-of-two-arrays.cpp
class Solution {
 public:
  vector<int> intersection(vector<int>& nums1, vector<int>& nums2)
  {
    unordered_map<int, bool> picked;
    for (int n : nums1) picked[n] = false;

    vector<int> ret;
    for (int n : nums2) {
      if (picked.find(n) != picked.end() && !picked[n]) {
        ret.push_back(n);
        picked[n] = true;
      }
    }

    return ret;
  }
};

File: leetcode/349-intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        picked = {}
        for n in nums1:
            picked[n] = False

        ret = []
        for n in nums2:
            if n in picked and not picked[n]:
                picked[n] = True
                ret.append(n)

        return ret
